crates/stages/src/stages/merkle.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    db::Transaction, ExecInput, ExecOutput, Stage, StageError, StageId, UnwindInput, UnwindOutput,
+};
+use reth_db::database::Database;
+use std::fmt::Debug;
+use tracing::*;
+
+const MERKLE_EXECUTION: StageId = StageId("MerkleExecuteStage");
+const MERKLE_UNWIND: StageId = StageId("MerkleUnwindStage");
+
+/// Merkle stage uses input from [AccountHashingStage] and [StorageHashingStage] stages
+/// and calculated intermediate hashed and state root.
+/// As input of this stage depends on Hashing stages this stage will be for execute be put after
+/// but for unwind before the hashing stages.
```

Comment:
```suggestion
/// This stage depends on the Account and Storage stages. It will be executed after them during execution, and before them during unwinding.
```